src/term/term_to_net.rs

```diff
@@ -174,6 +189,17 @@ fn encode_term(
 
       Ok(Some(Port(snd_node, 2)))
     }
+    Term::Tup { fst, snd } => {
+      let tup = inet.new_node(Con { lab: 1 });
```

Comment:
Is there som specific reason why you added labels to Con nodes? It seems that it only used here and the only used values are either 0 or 1.

I think you should remove the `lab` field and either compile to a Dup node like you did for Tup destructuring in the let case above (or use a Tup node type as I suggested).